op-supervisor/supervisor/backend/db/logs/db.go

```diff
@@ -301,6 +301,10 @@ func (db *DB) Contains(query types.ContainsQuery) (types.BlockSeal, error) {
 
 	evtHash, iter, err := db.findLogInfo(blockNum, logIdx)
 	if err != nil {
+		// if we get an ErrFuture but have a complete block, then we really have a conflict
+		if errors.Is(err, types.ErrFuture) && db.lastEntryContext.hasCompleteBlock() {
+			err = types.ErrConflict
+		}
```

Comment:
Could this logic be embedded all the way into `findLogInfo`? Seems like the same check could be applied to line 365, below. And that way the `findLogInfo` function doesn't need to specify that it could return "ErrFuture even if the block is fully known"